### PR TITLE
Fixing conf directory path to be taken from system property

### PR DESCRIPTION
**Pull request by himasha**

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/AndesConfigurationManager.java [hunk 1]
```diff
@@ -110,8 +111,10 @@
      * Main file path of configuration files into which all other andes-specific config files (if any)
      * should be linked.
      */
-    private static final String ROOT_CONFIG_FILE_PATH = System.getProperty(ServerConstants.CARBON_HOME) +
-            "/repository/conf/";
+    public static final String CARBON_CONFIG_DIR_PATH = "carbon.config.dir.path";
+    public static final String CARBON_HOME = "carbon.home";
+    private static  String ROOT_CONFIG_FILE_PATH;
```
Comment (hastef88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/AndesConfigurationManager.java]: Can make ROOT_CONFIG_FILE_PATH a non-static method level variable (given that its now a dynamic value.).

File: modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/AndesConfigurationManager.java [hunk 2]
```diff
@@ -145,7 +148,16 @@
      */
     public static void initialize(int portOffset) throws AndesException {
 
-        String brokerConfigFilePath = ROOT_CONFIG_FILE_PATH + ROOT_CONFIG_FILE_NAME;
+       // If system property is available set conf path to that value
+        if(componentsPath != null){
+            ROOT_CONFIG_FILE_PATH = Paths.get(componentsPath).toString();
+        }
+        else{
```
Comment (akalankapagoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/AndesConfigurationManager.java]: Needs to format this.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/AndesConfigurationManager.java (hunk 1 above)
Comment (himasha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/AndesConfigurationManager.java]: Fixed with 	79139c6

File: modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/AndesConfigurationManager.java (hunk 2 above)
Comment (himasha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/AndesConfigurationManager.java]: 79139c6

**Pull request merged**